source/core/module_typeHelpers.js

```diff
@@ -287,6 +293,8 @@
                 dispatcher: dispatchVisitArray
             },
             {
+                // Cluster is evaluated last, since internally Complex, AnalogWaveform and Timestamps
```

Comment:
We should add comments for the other types above where order is significant like checking String before Array, etc